src/main/java/uk/ac/sanger/sccp/stan/service/work/WorkServiceImp.java

```diff
@@ -193,6 +192,41 @@ public Work link(Work work, Collection<Operation> operations) {
         return workRepo.save(work);
     }
 
+    @Override
+    public void link(Collection<Work> works, Collection<Operation> operations) {
+        if (operations.isEmpty() || works.isEmpty()) {
+            return;
+        }
+        if (works.size()==1) {
+            link(works.iterator().next(), operations);
+            return;
+        }
+        List<String> inactiveWorkNumbers = works.stream()
+                .filter(work -> !work.isUsable())
+                .map(Work::getWorkNumber)
+                .collect(toList());
+        if (!inactiveWorkNumbers.isEmpty()) {
+            throw new IllegalArgumentException("Specified work cannot be used because it is not active: "+inactiveWorkNumbers);
+        }
+        Set<Integer> opIds = operations.stream().map(Operation::getId).collect(toLinkedHashSet());
+        Set<SampleSlotId> ssIds = operations.stream()
+                .flatMap(op -> op.getActions().stream()
+                        .map(a -> new SampleSlotId(a.getSample().getId(), a.getDestination().getId())))
+                .collect(toLinkedHashSet());
+
+        for (Work work : works) {
+            Set<Integer> workOpIds = new LinkedHashSet<>(work.getOperationIds());
+            workOpIds.addAll(opIds);
+            work.setOperationIds(new ArrayList<>(workOpIds));
+
+            Set<SampleSlotId> workSsIds = new LinkedHashSet<>(work.getSampleSlotIds());
+            workSsIds.addAll(ssIds);
+            work.setSampleSlotIds(new ArrayList<>(workSsIds));
```

Comment:
Also, just to understand why we are linking slot ids as well to work?

Reply:
We always link works to slots, just as a way of being to tell which labware and samples have been involved with which work numbers. It is not essential for the functionality of the service, but it can be useful in queries or for unknown future requirements.

Reply:
Great 👍